app/models/foreman_azure_rm/azure_rm.rb

```diff
@@ -188,40 +214,41 @@ def user_data_supported?
+      if args[:password].present? && !args[:ssh_key_data].present?
+        sudoers_cmd = "$echo #{args[:password]} | sudo -S echo '\"#{args[:username]}\" ALL=(ALL) NOPASSWD: ALL' > /etc/sudoers.d/waagent"
+        if args[:script_command].present?
           # to run the script_cmd given through form
           # as username
-          args[:azure_vm][:script_command] =  sudoers_cmd + " ; su - \"#{args[:azure_vm][:username]}\" -c \"#{args[:azure_vm][:script_command]}\""
+          args[:script_command] =  sudoers_cmd + " ; su - \"#{args[:username]}\" -c \"#{args[:script_command]}\""
         else
-          args[:azure_vm][:script_command] =  sudoers_cmd
+          args[:script_command] =  sudoers_cmd
         end
         disable_password_auth = false
-      elsif args[:azure_vm][:ssh_key_data].present? && !args[:azure_vm][:password].present?
+      elsif args[:ssh_key_data].present? && !args[:password].present?
         disable_password_auth = true
       else
         disable_password_auth = false
       end
+
       vm             = create_managed_virtual_machine(
-        name:                            args[:azure_vm][:vm_name],
+        name:                            args[:vm_name],
         location:                        args[:azure_vm][:location],
-        resource_group:                  args[:azure_vm][:resource_group],
-        vm_size:                         args[:azure_vm][:vm_size],
-        username:                        args[:azure_vm][:username],
-        password:                        args[:azure_vm][:password],
-        ssh_key_data:                    args[:azure_vm][:ssh_key_data],
+        resource_group:                  args[:resource_group],
+        vm_size:                         args[:vm_size],
+        username:                        args[:username],
+        password:                        args[:password],
+        ssh_key_data:                    args[:ssh_key_data],
         disable_password_authentication: disable_password_auth,
         network_interface_card_ids:      nics.map(&:id),
-        platform:                        args[:azure_vm][:platform],
+        platform:                        args[:platform],
         vhd_path:                        args[:image_id],
-        os_disk_caching:                 args[:azure_vm][:os_disk_caching],
-        premium_os_disk:                 args[:azure_vm][:premium_os_disk],
+        os_disk_caching:                 args[:os_disk_caching],
+        premium_os_disk:                 args[:premium_os_disk],
         custom_data:                     args[:user_data],
-        script_command:                  args[:azure_vm][:script_command],
-        script_uris:                     args[:azure_vm][:script_uris],
+        script_command:                  args[:script_command],
+        script_uris:                     args[:script_uris],
       )
       create_vm_extension(args)
       # return the vm object using azure_vm
```

Comment:
Field `resource_group` inside the `base` erb defined using your custom class object. 
So in below line, don't you need to assign `resource_group` for `AzureRMCompute` instance?

Reply:
handling for `create_vm` scenario now.